### ISSUE TITLE
Concurrent republishing ? 

### ISSUE THREAD

**Issue by MichaelMure** [ipfs/go-ipfs-provider#17]
Unless I missed something, this package does only sequential republishing and thus is quite slow.

Would it make sense to have concurrent republishing ? Any constraint I should be aware of ?

**Comment by MichaelMure** [comment]
cc @Stebalien

**Comment by MichaelMure** [comment]
For some context, I bolted together this package, a block/datastore and the DHT, I get about 1 key published per minute.

**Comment by Stebalien** [comment]
> I get about 1 key published per minute

It's hitting a timeout. This is why we're working on fixing the DHT.

> Would it make sense to have concurrent republishing ? Any constraint I should be aware of ?

Maybe but DHT queries internally create several connections. It might make sense to perform, say, 2-8 at a time, but I wouldn't do more than that for now.

**Comment by hacdias** [comment]
This repository is no longer maintained and has been copied over to [Boxo](https://github.com/ipfs/boxo).  In an effort to avoid noise and crippling in the Boxo repo from the weight of issues of the past, we are closing most issues and PRs in this repo.  Please feel free to open a new issue in Boxo (and reference this issue) if resolving this issue is still critical for unblocking or improving your usecase.

You can learn more in the [FAQs for the Boxo repo copying/consolidation effort](https://github.com/ipfs/boxo/wiki/Copied-or-Migrated-Repos-FAQ).
